### DISCUSSION TITLE
Adding a openupm pacakge as dependency

### DISCUSSION THREAD

**Discussion by tratteo** [openupm/openupm#3293] [Q&A] [answered]
Good evening, I am quite new to OpenUPM.
Is it possible to add as a dependency for a package another OpenUPM package?
For clarity, I would need the package **com.tratteo.pury** to be a dependency of **com.tratteo.gibframe**, both are released on the OpenUPM registry. 
For now i tried adding the line 
`com.tratteo.pury: "1.0.0"` inside the dependencies in the package.json without any result. Thanks

**Comment by favoyang** [comment, marked as answer]
Yes, you can. But you need to add a scope registry with "com.tratteo" scope points to the openupm registry. Or if you're using openupm-cli, it will do it for you.

We have some docs on the website you can checkout.

**Reply by tratteo** [reply]
Thanks for the reply. Do I have to create a new release in order to update the dependencies?

**Reply by favoyang** [reply]
Yes. As the service only tracks git tags, you will need to create a new version for it.

However, I suggest you import the package locally to test the dependency, then commit to git and create a git tag for it.

**Reply by tratteo** [reply]
One last question: is it possible to add as a dependency a package automatically specifying the "latest" version available as the required one (without specifying the number)?

**Reply by favoyang** [reply]
If you mean automatically installing the latest dependency when installing the source package. Then, no. The UPM system requires strong deterministic at the moment and only supports a fixed version.

Didn't like NPM which gives you some sort of flexibility with tilde and caret: https://michaelsoolee.com/npm-package-tilde-caret/
